tests/playwright-test/test-modifiers.spec.ts

```diff
@@ -214,3 +214,74 @@ test('modifier with a function should throw in the test', async ({ runInlineTest
   expect(result.exitCode).toBe(1);
   expect(result.output).toContain('test.skip() with a function can only be called inside describe block');
 });
+
+test('test.skip with worker fixtures only should skip before hooks and tests', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'a.test.ts': `
+      const test = pwt.test.extend({
+        bar: 'bar',
+        foo: [ 'foo', { scope: 'worker' }],
+      });
+      const logs = [];
+      test.beforeEach(() => {
+        console.log('\\n%%beforeEach');
+      });
+      test('passed', () => {
+        console.log('\\n%%passed');
+      });
+      test.describe('suite1', () => {
+        test.skip(({ foo }) => foo === 'foo', 'reason');
```

Comment:
I'd expect a log here. How are we testing that this is run only once? I assume that the old behavior had the beforeEach running for each skip? I didn't realize that beforeEach ran before skips in the first place.

Reply:
The old behavior was to run skip for each test, now we only run it once when possible, before any beforeAll.